pyface/ui/qt4/single_choice_dialog.py

```diff
@@ -60,7 +60,10 @@ def _create_contents(self, parent):
 
     def _show_modal(self):
         self.control.setWindowModality(QtCore.Qt.WindowModality.ApplicationModal)
-        retval = self.control.exec_()
+        if qt_api == 'pyside2':
+            retval = self.control.exec_()
+        else:
+            retval = self.control.exec()
```

Comment:
any particular reason why we're not using `hasattr` here?

Reply:
No - I did it that way first and then converted to hasattr but I clearly missed one.